src/jaxsim/rbda/contacts/relaxed_rigid.py

```diff
@@ -0,0 +1,361 @@
+                )
+            }
+        )
+
+    def valid(self) -> bool:
+        return bool(
+            jnp.all(self.time_constant >= 0.0)
+            and jnp.all(self.damping_coefficient > 0.0)
+            and jnp.all(self.d_min >= 0.0)
+            and jnp.all(self.d_max <= 1.0)
+            and jnp.all(self.d_min <= self.d_max)
+            and jnp.all(self.width >= 0.0)
+            and jnp.all(self.midpoint >= 0.0)
+            and jnp.all(self.power >= 0.0)
+            and jnp.all(self.mu >= 0.0)
+            and jnp.all(self.max_iterations > 0)
+            and jnp.all(self.tolerance > 0.0)
+        )
+
+
+@jax_dataclasses.pytree_dataclass
+class RelaxedRigidContactsState(ContactsState):
+    """Class storing the state of the relaxed rigid contacts model."""
+
+    def __eq__(self, other: RelaxedRigidContactsState) -> bool:
+        return hash(self) == hash(other)
+
+    @staticmethod
+    def build() -> RelaxedRigidContactsState:
+        """Create a `RelaxedRigidContactsState` instance"""
+
+        return RelaxedRigidContactsState()
+
+    @staticmethod
+    def zero(model: js.model.JaxSimModel) -> RelaxedRigidContactsState:
+        """Build a zero `RelaxedRigidContactsState` instance from a `JaxSimModel`."""
+        return RelaxedRigidContactsState.build()
+
+    def valid(self, model: js.model.JaxSimModel) -> bool:
+        return True
+
+
+@jax_dataclasses.pytree_dataclass
+class RelaxedRigidContacts(ContactModel):
+    """Rigid contacts model."""
+
+    parameters: RelaxedRigidContactsParams = dataclasses.field(
+        default_factory=RelaxedRigidContactsParams
+    )
+
+    terrain: jax_dataclasses.Static[Terrain] = dataclasses.field(
+        default_factory=FlatTerrain
+    )
+
+    def compute_contact_forces(
+        self,
+        position: jtp.Vector,
+        velocity: jtp.Vector,
+        model: js.model.JaxSimModel,
+        data: js.data.JaxSimModelData,
+    ) -> tuple[jtp.Vector, tuple[Any, ...]]:
+
+        def _detect_contact(x: jtp.Array, y: jtp.Array, z: jtp.Array) -> jtp.Array:
+            x, y, z = jax.tree_map(jnp.squeeze, (x, y, z))
+
+            n̂ = self.terrain.normal(x=x, y=y).squeeze()
+            h = jnp.array([0, 0, z - model.terrain.height(x=x, y=y)])
+
+            return jnp.dot(h, n̂)
+
+        # Compute the activation state of the collidable points
+        δ = jax.vmap(_detect_contact)(*position.T)
+
+        with data.switch_velocity_representation(VelRepr.Mixed):
+            M = js.model.free_floating_mass_matrix(model=model, data=data)
+            Jl_WC = jnp.vstack(
+                jax.vmap(lambda J, height: J * (height < 0))(
+                    js.contact.jacobian(model=model, data=data)[:, :3, :], δ
+                )
+            )
+            W_H_C = js.contact.transforms(model=model, data=data)
+            W_ν̇ = jnp.hstack(js.ode.system_acceleration(model=model, data=data))
+            W_ν = data.generalized_velocity()
```

Comment:
Sorry to bother once again :)

```suggestion
            W_ν̇ = jnp.hstack(js.ode.system_acceleration(model=model, data=data))
            BW_ν = data.generalized_velocity()
```

While you can get the mixed generalized velocity ${}^{B[W]} \boldsymbol{\nu}$ within a context manager, this [does not apply](https://github.com/ami-iit/jaxsim/blob/58b803c426d049232afe74803e6c3db380f31f87/src/jaxsim/api/ode.py#L286) for historical reasons to its derivative (we can consider to update this old logic). It seems to me that in your logic, you need both expressed in mixed. The way you do it, you would not get $\dot{\boldsymbol{\nu}}$ in mixed representation.

Reply:
Actually, this is a new method introduced by @xela-95. It was split from the `system_velocity_dynamics` in #227. Since it calls ABA under the hood, and ABA already supports returning base accelerations in all the supported representations, probably it may make sense to update the `system_acceleration` method such that it returns the acceleration in the active representation of its `data` argument. What do you think @xela-95? The original constrain of returning an inertial-fixed acceleration is only necessary in `system_velocity_dynamics` since we consider the ODE in inertial-fixed representation.

Reply:
Addressed in:
- `ode` module in https://github.com/ami-iit/jaxsim/pull/223/commits/32d5293873f56ef34d2fe7ad826a350d06317adf
- rigid contacts in https://github.com/ami-iit/jaxsim/pull/223/commits/32d5293873f56ef34d2fe7ad826a350d06317adf
- relaxed rigid contact in https://github.com/ami-iit/jaxsim/pull/223/commits/43e50d2beef8b6c8a527c663414c878415fff6de

Reply:
> Actually, this is a new method introduced by @xela-95. It was split from the `system_velocity_dynamics` in #227. Since it calls ABA under the hood, and ABA already supports returning base accelerations in all the supported representations, probably it may make sense to update the `system_acceleration` method such that it returns the acceleration in the active representation of its `data` argument. What do you think @xela-95? The original constraint of returning an inertial-fixed acceleration is only necessary in `system_velocity_dynamics` since we consider the ODE in inertial-fixed representation.

Yes I agree, thanks for having pointed out this!